pilot/pkg/model/push_context.go

```diff
@@ -979,6 +1052,44 @@ func (ps *PushContext) initDestinationRules(env *Environment) error {
 	return nil
 }
 
+// AuthenticationPolicyForWorkload returns the matching auth policy for a given service
+// This replaces store.AuthenticationPolicyForWorkload
+func (ps *PushContext) AuthenticationPolicyForWorkload(service *Service, port *Port) *authn.Policy {
+	var workloadPolicy *authn.Policy
+	// Match by Service hostname
+	if workloadPolicy = authenticationPolicyForWorkload(ps.AuthnPolicies.policies[service.Hostname], port); workloadPolicy != nil {
+		return workloadPolicy
+	}
+
+	// Match by namespace
+	if workloadPolicy = authenticationPolicyForWorkload(ps.AuthnPolicies.policies[host.Name(service.Attributes.Namespace)], port); workloadPolicy != nil {
+		return workloadPolicy
+	}
+
+	// Use default global authentication policy if no others found
+	return ps.AuthnPolicies.defaultMeshPolicy
+}
+
+func authenticationPolicyForWorkload(policiesByPort []*authnPolicyByPort, port *Port) *authn.Policy {
+	var matchedPolicy *authn.Policy
+	if policiesByPort == nil {
```

Comment:
> Or are you questioning the validity of the 1st check on line 1080. I added that because I wasn't sure if there would ever be a scenario where port could be nil

i haven't followed up much about the relationship with "better default networking" vs "authenticaiton policy". @diemtvu @lambdai with new `capture-all` mode for inbound traffic, if I'm defining a mTLS authentication policy without specifying port for a service, should I expect all inbound traffic is treated as mTLS enabled? I would expect so, but not sure.

If so, it's could it be possible this is invoked with `nil` port?

Just want to ensure we're keep all ends tight.

Reply:
I filed https://github.com/istio/istio/issues/17278 to track separately.